themes/_core/js/cart.js

```diff
@@ -61,34 +61,34 @@ $(document).ready(() => {
 
   var $body = $('body');
 
-  $body.on(
-    'click',
-    '[data-button-action="add-to-cart"]',
-    (event) => {
-      event.preventDefault();
-
-      var $form = $($(event.target).closest('form'));
-      var query = $form.serialize() + '&add=1&action=update';
-      var actionURL = $form.attr('action');
+$body.on(
+  'click',
+  '[data-button-action="add-to-cart"]',
+  (event) => {
+    if ($('#quantity_wanted').val() > $('[data-stock]').data('stock') && $('[data-allow-oosp]').data('allow-oosp').length === 0) {
+        $('[data-button-action="add-to-cart"]').attr('disabled', 'disabled');
```

Comment:
Is there a server-side validation to make sure you cannot add the item regardless of front-end checks?